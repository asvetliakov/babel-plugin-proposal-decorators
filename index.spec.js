const { transform } = require("@babel/core");
const { default: plugin } = require("./");

const source = `
function TestDec1() {
    return (target, propertyKey, description) => { }
}

function TestDec2() {
    return (target, propertyKey, description) => { }
}

class A {
    @TestDec1()
    prop1 = 5;
    @TestDec2()
    func() { }
}
`;

it("Works", () => {
    expect(transform(source, {
        presets: [
            ["@babel/stage-3", { loose: true }],
            "@babel/es2017",
            "@babel/es2016",
        ],
        plugins: [
            plugin
        ]
    }).code).toMatchSnapshot();
});