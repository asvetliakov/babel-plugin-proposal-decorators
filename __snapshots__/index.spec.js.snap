// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works 1`] = `
"var _dec, _dec2, _class, _descriptor;

function _initializerDefineProperty(target, property, descriptor, context) { if (!descriptor) return; Object.defineProperty(target, property, { enumerable: descriptor.enumerable, configurable: descriptor.configurable, writable: descriptor.writable, value: descriptor.initializer ? descriptor.initializer.call(context) : void 0 }); }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) { var desc = {}; Object['ke' + 'ys'](descriptor).forEach(function (key) { desc[key] = descriptor[key]; }); desc.enumerable = !!desc.enumerable; desc.configurable = !!desc.configurable; desc.writable = true; desc = decorators.slice().reverse().reduce(function (desc, decorator) { return decorator(target, property, desc) || desc; }, desc); if (context && desc.initializer !== void 0) { desc.value = desc.initializer ? desc.initializer.call(context) : void 0; desc.initializer = undefined; } if (desc.get || desc.set) { delete desc.writable; } if (desc.initializer === void 0) { Object['define' + 'Property'](target, property, desc); desc = null; } return desc; }

function _initializerWarningHelper(descriptor, context) { throw new Error('Decorating class property failed. Please ensure that ' + 'proposal-class-properties is enabled and set to use loose mode. ' + 'To use proposal-class-properties in spec mode with decorators, wait for ' + 'the next major version of decorators in stage 2.'); }

function TestDec1() {
  return (target, propertyKey, description) => {};
}

function TestDec2() {
  return (target, propertyKey, description) => {};
}

let A = (_dec = TestDec1(), _dec2 = TestDec2(), (_class = class A {
  constructor() {
    _initializerDefineProperty(this, \\"prop1\\", _descriptor, this);
  }

  func() {}

}, (_descriptor = _applyDecoratedDescriptor(_class.prototype, \\"prop1\\", [_dec], {
  enumerable: true,
  initializer: function () {
    return 5;
  }
}), _applyDecoratedDescriptor(_class.prototype, \\"func\\", [_dec2], Object.getOwnPropertyDescriptor(_class.prototype, \\"func\\"), _class.prototype)), _class));"
`;
